operator/builtin/transformer/recombine/recombine_test.go

```diff
@@ -141,6 +149,27 @@ func TestRecombineOperator(t *testing.T) {
 			},
 			[]*entry.Entry{entryWithBody(t1, "test1test2")},
 		},
+		{
+			"fourEntriesTwoSourceIdentifier",
+			func() *RecombineOperatorConfig {
+				cfg := NewRecombineOperatorConfig("")
+				cfg.CombineField = entry.NewBodyField()
+				cfg.IsLastEntry = "$body == 'end'"
+				cfg.OutputIDs = []string{"fake"}
+				cfg.SourceIdentifier = entry.NewAttributeField("file.path")
+				return cfg
+			}(),
+			[]*entry.Entry{
+				entryWithBodyAttr(t1, "file1", map[string]string{"file.path": "file1"}),
+				entryWithBodyAttr(t1, "file2", map[string]string{"file.path": "file2"}),
+				entryWithBodyAttr(t2, "end", map[string]string{"file.path": "file1"}),
+				entryWithBodyAttr(t2, "end", map[string]string{"file.path": "file2"}),
+			},
+			[]*entry.Entry{
+				entryWithBodyAttr(t1, "file1\nend", map[string]string{"file.path": "file1"}),
+				entryWithBodyAttr(t1, "file2\nend", map[string]string{"file.path": "file2"}),
+			},
+		},
```

Comment:
Would you be willing to add a couple more test cases? 
- That `DefaultSourceIdentifier` works as expected
- That `max_sources` works as expected
- That `max_batch_size` works as expected (Not part of your changes, but would be appreciated since it's related)

Reply:
sounds good. added.